_tutorials/new-component/create-a-new-component-part-7.md

```diff
@@ -0,0 +1,141 @@
+
+## SassDoc
+
+To document Sass functions, mixins, and variables Origami uses [SassDoc](http://sassdoc.com/). A SassDoc comment comes before the Sass to document. It uses three forward slashes and has a number of [annotations](sassdoc.com/annotations/). Its easier to see with an example. Let's add SassDoc comments to our undocumented `oExampleAddTheme` mixin we created earlier:
+
+<pre><code class="o-syntax-highlight--scss">/// Output a css modifier class for an o-example theme.
+/// This may be an existing theme, or a new custom them.
+/// @param {String} $name - The name of the theme to output.
+/// @param {Map} $opts [null] - A map of theme options to create a custom theme. See the examples and README for full details.
+/// @example scss - Output a custom theme modifier class "o-example--my-custom-theme".
+///     @include oExampleAddTheme('my-custom-theme', (
+///     	'background-color': oColorsByName('white'),
+///     	'text-color': oColorsByName('crimson'),
+///     	'button-color': oColorsByName('crimson'),
+///     ));
+/// @example scss - Output a default o-example theme modifier class "o-example--inverse".
+///     @include oExampleAddTheme('inverse');
+/// @access public
+@mixin oExampleAddTheme($name, $opts: null) {
+    // theme sass here from before, see part 3
+}
+</code></pre>
+
+Above we give a [description](http://sassdoc.com/annotations/#description) of the mixins usecase. We use the [`@param`](http://sassdoc.com/annotations/#parameter) annotation to document the parameters of our mixin. Note that SassDoc does not provide a way to document each property of the `$opts` map. Instead we use the [`@example`](http://sassdoc.com/annotations/#example) annotation and readme to compensate. Finally we use the [`@access`](http://sassdoc.com/annotations/#access) annotation to indicate the mixin is public and can be included by component users. Although our use of `@access` is superfluous, as [only Sass which begins with an underscore is considered private](/spec/v1/sass/#private-sass).
+
+<figure>
+	<img alt="" src="/assets/images/tutorial-new-component/hello-world-demo-17-docs.png" />
+	<figcaption class="o-typography-caption">
+        The SassDoc is shown in the Origami registry. This image shows <a href="https://registry.origami.ft.com/components/o-table@8.1.2/sassdoc?brand=master">o-table SassDocs</a>.
+	</figcaption>
+</figure>
+
+## JSDoc
+
+To document JavaScript Origami components use [JSDoc comments](https://jsdoc.app/about-getting-started.html). For example see the existing JSDoc for the `init` method of our `o-example` component, which was generated for us by the `obt init` command:
```

Comment:
```suggestion
To document JavaScript, Origami components use [JSDoc comments](https://jsdoc.app/about-getting-started.html). For example, see the existing JSDoc for the `init` method of our `o-example` component, which was generated for us by the `obt init` command:
```

Reply:
> Maybe instead of "To document JavaScript Origami components use JSDoc comments" this should be "Origami components use JSDoc comments to document JavaScript"?

Yes, I much prefer "Origami components use JSDoc comments to document JavaScript" 😄